### ISSUE TITLE
Turn the Event to a class or TypedDict

### ISSUE THREAD

**Issue by xlurio** [getsentry/sentry-python#2014]
### Problem Statement

Currently, the `Event` type is a dictionary with a non documented structure and because of that it makes harder to preprocess events using `before_send`.

### Solution Brainstorm

Defining a class or [TypedDict](https://peps.python.org/pep-0589/) would make it to have a well defined interface, which would turn easier for developers to know how to access or change the desired data.

**Comment by antonpirker** [comment]
Hey @xlurio !

Good idea! We just need to make sure that it also works in Python 2 (we still support Python 2.7)
So I guess just applying these types in Python 3.

Als one concern is that we can break existing implementations of `before_send` that assume that some is in the event/hint that is not specified in the type. (We have to be more careful the more users we have, because everything is used and missused :-) )

**Comment by antonpirker** [comment]
Would you be willing to come up with a PR for that where our test suite is running successful?

**Comment by xlurio** [comment]
> Hey @xlurio !
> 
> Good idea! We just need to make sure that it also works in Python 2 (we still support Python 2.7) So I guess just applying these types in Python 3.
> 
> Als one concern is that we can break existing implementations of `before_send` that assume that some is in the event/hint that is not specified in the type. (We have to be more careful the more users we have, because everything is used and missused :-) )

Yeah, that is why I thought on TypedDict, since changing a dictionary by class would probably cause a breaking change. TypedDict would only affect the type stubs so it shouldn't break any existing code.



> Would you be willing to come up with a PR for that where our test suite is running successful?

I'll be glad to!

**Comment by sentrivana** [comment]
Closing in favor of https://github.com/getsentry/sentry-python/issues/2357
